src/odemis/gui/cont/tabs.py

```diff
@@ -508,20 +508,30 @@ def load_data(self, data):
         # Create streams from data
         streams = data_to_static_streams(data)
 
-        # TODO: Clear previous overview streams
-        self.clear_data()
-
         for s in streams:
+            self.clear_data(s)
             scont = self._overview_stream_controller.addStream(s, add_to_view=True)
             scont.stream_panel.show_remove_btn(True)
 
-    def clear_data(self):
+    def clear_data(self, new_stream=None):
         """
         Clear the tab data upon resetting the project:
         - Clear overview map streams
-        - Clear live streams data
-        """
-        self._overview_stream_controller.clear()
+        @:param new_stream: (StaticStream or None) the newly acquired overview stream
+        """
+        # Clear old static streams from the view
```

Comment:
I think that the interface is fine, but the naming is confusing. I find it hard to guess what will happen when calling `clear_data(my_shiny_new_stream)`. It feels like it's going to remove that new stream. 
I'd suggest something like `clear_static_data(filter=None)`. `filter` could be a "similar stream", but I'd go for an even simpler behaviour and just pass a string: the name of the stream(s) to remove. However, `filter` is not a great name, as it's a builtin of Python. Also, when changing of selected Feature, we'll need a similar behaviour which removes all the static streams from the previous feature. So maybe we need to extend the signature to handle such behaviour, or alternatively we'd have a `clear_overview_data(...)` and a `clear_acquired_data(...)`.
In the docstring, please indicate what happens when passing `None`.

Also, we should have a counter-part `clear_live_data()`, which takes care of resetting the live streams when a new project is started. It should just empty the content of the live streams (so that the last image of the previous sample are not shown at the beginning)